### PR TITLE
Ko datadir

### PR DESCRIPTION
**Pull request by greghaynes**

### CODE REVIEW

**Comment by greghaynes** [comment]
/hold

This depends on #22 and a fix in manifestival

File: cmd/manager/kodata/knative-serving-0.6.0.yaml [hunk 1]
```diff
@@ -0,0 +1 @@
+../../../config/resources/knative-serving-0.6.0.yaml
```
Comment (jcrossley3) [inline cmd/manager/kodata/knative-serving-0.6.0.yaml]: Any problem symlinking to all of `resources/`? There are other manifests in there for setting up a minimal istio controlplane and automatic ingress routes on openshift.

**Comment by trshafer** [comment]
The change to manifestival has been added: https://github.com/jcrossley3/manifestival/commit/78b6b04ae6ff35401397219bbbf0e97cd349712e

File: pkg/controller/knativeserving/knativeserving_controller.go [hunk 2]
```diff
@@ -104,8 +104,10 @@ type ReconcileKnativeServing struct {
 
 // Create manifestival resources and KnativeServing, if necessary
 func (r *ReconcileKnativeServing) InjectClient(c client.Client) error {
-	m, err := mf.NewManifest(*filename, *recursive, c)
+	koDataDir := os.Getenv("KO_DATA_PATH")
+	m, err := mf.NewManifest(filepath.Join(koDataDir, "knative-serving-0.6.0.yaml"), *recursive, c)
 	if err != nil {
+		log.Error(err, "Failed to load manifest")
 		return err
```
Comment (houshengbo) [inline pkg/controller/knativeserving/knativeserving_controller.go]: Plz move deploy/crds/serving_v1alpha1_knativeserving_cr.yaml to  cmd/manager/kodata/ as well.

And change func (r *ReconcileKnativeServing) ensureKnativeServing() (err error) in this file.

File: cmd/manager/kodata/knative-serving-0.6.0.yaml (hunk 1 above)
Comment (houshengbo) [reply cmd/manager/kodata/knative-serving-0.6.0.yaml]: Better have the file on a one-on-one basic. We need file under other folder as well, like config/crds/serving_v1alpha1_knativeserving_cr.yaml

**Comment by evankanderson** [comment]
Why the large changes to `Gopkg.lock`?

**Comment by houshengbo** [comment]
Gopkg.lock generated by running ./hack/update-deps.sh. There must have been some changes in Gopkg.toml, so that new Gopkg.lock is updated.

**Comment by trshafer** [comment]
If you rebase, that will pull in #24 which should fix the large change to the Gopkg files.

**Comment by greghaynes** [comment]
I had to do an update-deps to pull in something new which also updated the out of sync deps we already had. Rebasing should fix this

File: cmd/manager/kodata/knative-serving-0.6.0.yaml (hunk 1 above)
Comment (greghaynes) [reply cmd/manager/kodata/knative-serving-0.6.0.yaml]: Yes IMO id be explicit, alternatively I could make a second subdir within kodir and link that to config but as is Id expect this to work just fine.

**Comment by trshafer** [comment]
Hey @greghaynes, I have create a PR that incorporates the fix @houshengbo requested. https://github.com/greghaynes/serving-operator/pull/1 

I have verified that this fix installs knative-serving: https://gist.github.com/trshafer/8ef782b731218fe410bdc35498c56adb.

File: cmd/manager/kodata/serving_v1alpha1_knativeserving_cr.yaml [hunk 3]
```diff
@@ -0,0 +1 @@
+serving_v1alpha1_knativeserving_cr.yaml
```
Comment (mattmoor-sockpuppet) [inline cmd/manager/kodata/serving_v1alpha1_knativeserving_cr.yaml]:
```suggestion
serving_v1alpha1_knativeserving_cr.yaml

```

File: cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml [hunk 4]
```diff
@@ -0,0 +1 @@
+serving_v1alpha1_knativeserving_crd.yaml
```
Comment (mattmoor-sockpuppet) [inline cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml]:
```suggestion
serving_v1alpha1_knativeserving_crd.yaml

```

File: cmd/manager/kodata/serving_v1alpha1_knativeserving_cr.yaml (hunk 3 above)
Comment (mattmoor-sockpuppet) [inline cmd/manager/kodata/serving_v1alpha1_knativeserving_cr.yaml]:
```suggestion
serving_v1alpha1_knativeserving_cr.yaml

```

File: cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml (hunk 4 above)
Comment (mattmoor-sockpuppet) [inline cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml]:
```suggestion
serving_v1alpha1_knativeserving_crd.yaml

```

**Comment by greghaynes** [comment]
@trshafer on nice - if we can lets merge this then you can rebase your change on master and we'll merge your PR. Otherwise we have to manually poke CLA bot and it gets messy.

**Comment by trshafer** [comment]
Hey @greghaynes, I'm totally happy if you want to take the changes I made and squash them into your commit. Feel free to close my PR. I'm also happy to take my change and make a new PR on this repo after this PR is merged. Whatever is easiest for you.

File: cmd/manager/kodata/serving_v1alpha1_knativeserving_cr.yaml (hunk 3 above)
Comment (houshengbo) [inline cmd/manager/kodata/serving_v1alpha1_knativeserving_cr.yaml]: This is a broken link. You need to link the same way as 
../../../config/crds/serving_v1alpha1_knativeserving_cr.yaml.

File: cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml (hunk 4 above)
Comment (houshengbo) [inline cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml]: This is a broken link. You need to link the same way as knative-serving-0.6.0.yaml, like
../../../config/crds/serving_v1alpha1_knativeserving_cr.dyaml.

Comment (trshafer) [reply cmd/manager/kodata/serving_v1alpha1_knativeserving_crd.yaml]: I think only the cr is necessary.

**Pull request merged**